### PR TITLE
Add missing methods for MVP

### PR DESCRIPTION
**Pull request by pmerlet-at-didomi**

### CODE REVIEW

File: example/lib/widgets/check_consent.dart [hunk 1]
```diff
@@ -15,17 +15,22 @@ class _CheckConsentState extends BaseSampleWidgetState<CheckConsent> {
...
-    return 'Result = $result';
+    final bool consentRequired = await DidomiSdk.isConsentRequired;
+    final bool shouldConsentBeCollected = await DidomiSdk.shouldConsentBeCollected;
+    final bool consentStatusPartial = await DidomiSdk.isUserConsentStatusPartial;
+    final bool liStatusPartial = await DidomiSdk.isUserLegitimateInterestStatusPartial;
+    return '\n- Is user consent required? $consentRequired\n'
+        '- Should consent be collected? $shouldConsentBeCollected\n'
+        '- Is status partial? consent => $consentStatusPartial, LI => $liStatusPartial';
   }
 }
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/check_consent.dart:36]: missing blank line

File: example/lib/widgets/show_hide_preferences.dart [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            });
+          }
+        },
+        controlAffinity:
+        ListTileControlAffinity.leading, //  <-- leading Checkbox
+      ),
+      buildResponseText(getActionId())
+    ];
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/show_hide_preferences.dart:136]: missing empty line

File: lib/preferences_view.dart [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/// Possible opened pages when calling showPreferences
+enum PreferencesView {
+  /// Main preferences screen
+  purposes,
+  /// Vendors preferences screen
+  vendors
+}
```
Comment (nicolas-chaix-didomi) [inline lib/preferences_view.dart:7]: missing empty line

**Review by nicolas-chaix-didomi — CHANGES_REQUESTED** [review]

File: lib/preferences_view.dart (hunk 3 above)
Comment (pmerlet-at-didomi) [reply lib/preferences_view.dart]: Fixed

File: example/lib/widgets/show_hide_preferences.dart (hunk 2 above)
Comment (pmerlet-at-didomi) [reply example/lib/widgets/show_hide_preferences.dart]: Fixed

File: example/lib/widgets/check_consent.dart (hunk 1 above)
Comment (pmerlet-at-didomi) [reply example/lib/widgets/check_consent.dart]: Fixed

**Review by nicolas-chaix-didomi — APPROVED** [review]

File: ios/Classes/SwiftDidomiSdkPlugin.swift [hunk 4]
```diff
@@ -358,4 +388,54 @@ public class SwiftDidomiSdkPlugin: NSObject, FlutterPlugin {
...
+            return
+        }
+        guard let organizationUserIdAuthDigest = argumentOrError(argumentName: "organizationUserIdAuthDigest", methodName: "setUserWithAuthentication", args: args, result: result) else {
+            return
+        }
+        Didomi.shared.setUser(
+            id: userId,
+            algorithm: organizationUserIdAuthAlgorithm,
+            secretId: organizationUserIdAuthSid,
+            salt: args["organizationUserIdAuthSalt"] as? String ?? nil,
```
Comment (felipe-saez) [inline ios/Classes/SwiftDidomiSdkPlugin.swift:427]: since salt is an optional, you should be able to remove ` ?? nil`

**Review by felipe-saez — APPROVED** [review]

**Pull request merged**